File: chap21/C#/chap21/CADDialog.cs
using System;
using System.Collections.Generic;
using System.Text;
using Autodesk.AutoCAD.Runtime;
using Autodesk.AutoCAD.ApplicationServices;
namespace chap21
{
    public class CADDialog
    {
        [CommandMethod("ModalForm")]
        public void ShowModalForm()
        {
            //显示模态对话框
            ModalForm modalForm =new ModalForm();
            Application.ShowModalDialog(modalForm);
        }
    }
}

File: chap21/VB/chap21/CADDialog.vb
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports Autodesk.AutoCAD.Runtime
Imports Autodesk.AutoCAD.ApplicationServices
Public Class CADDialog
<CommandMethod("ModalForm")> _
Public Sub ShowModalForm()
    '显示模态对话框
    Dim modalForm As New ModalForm()
    Application.ShowModalDialog(modalForm)
End Sub
End Class
